File: app/src/main/java/com/microsoft/msftbreweryworkshop/service/BreweryService.java
package com.microsoft.msftbreweryworkshop.service;

import com.microsoft.msftbreweryworkshop.api.model.BreweryDetail;
import com.microsoft.msftbreweryworkshop.api.model.BreweryItem;
import com.microsoft.msftbreweryworkshop.api.model.DetailResponse;
import com.microsoft.msftbreweryworkshop.api.model.ListResponse;

public interface BreweryService {
    void getBreweries(ServiceListener<ListResponse<BreweryItem>> listener);
    void getBrewery(String id, ServiceListener<DetailResponse<BreweryDetail>> listener);
}

File: app/src/main/java/com/microsoft/msftbreweryworkshop/service/BreweryService.kt
package com.microsoft.msftbreweryworkshop.service

import com.microsoft.msftbreweryworkshop.api.model.BreweryDetail
import com.microsoft.msftbreweryworkshop.api.model.BreweryItem
import com.microsoft.msftbreweryworkshop.api.model.DetailResponse
import com.microsoft.msftbreweryworkshop.api.model.ListResponse

interface BreweryService {
    fun getBreweries(listener: ServiceListener<ListResponse<BreweryItem>>)
    fun getBrewery(id: String, listener: ServiceListener<DetailResponse<BreweryDetail>>)
}